src/store/providers/history-provider.js

```diff
@@ -0,0 +1,102 @@
+import { createContext, useContext, useState } from "react"
+import axios from "axios"
+import { useNavigate, useLocation } from 'react-router-dom'
+import { useToast } from 'react-felix-ui'
+
+const HistoryContext = createContext()
+
+const initState = {
+    history: [],
+    count: 0,
+    // playTime: 0,
+}
+const HistoryProvider = ({ children }) => {
+
+    const [HistoryState, setHistoryState] = useState(initState)
+    const encodedToken = localStorage.getItem("felix-tv-user-token");
+    const toast = useToast()
+    const navigate = useNavigate()
+    const location = useLocation()
+
+    const addToHistory = (item) => {
+
+        if (!checkInHistory(item._id)) {
+            axios.post("/api/user/history",
+                { video: item },
+                {
+                    headers: {
+                        authorization: encodedToken,
+                    },
+                }
+            ).then((response) => {
+                setHistoryState({ ...HistoryState, history: response.data.history });
+            }).catch((err) => {
+                toast({
+                    status: "error",
+                    message: "Sign in to your account first",
+                    duration: 2
+                })
+                navigate('/signin', { state: { from: location }, replace: true })
+            })
+        }
+    }
+
+    const removeFromHistory = (id) => {
+        axios.delete(`/api/user/history/${id}`, {
+            headers: {
+                authorization: encodedToken,
```

Comment:
suggestion:  to use interceptors to add token in one place

Reply:
Not sure what you ar suggesting